source/extensions/tracers/xray/config.cc

```diff
@@ -29,8 +33,18 @@ XRayTracerFactory::createHttpTracerTyped(const envoy::config::trace::v2::XRayCon
     ENVOY_LOG(error, "Failed to read sampling rules manifest because of {}.", e.what());
   }
 
-  XRayConfiguration xconfig{proto_config.daemon_endpoint(), proto_config.segment_name(),
-                            sampling_rules_json};
+  RELEASE_ASSERT(proto_config.daemon_endpoint().protocol() ==
+                     api::core::SocketAddress::Protocol::SocketAddress_Protocol_UDP,
+                 "X-Ray daemon endpoint must be a UDP socket address");
+
+  RELEASE_ASSERT(proto_config.daemon_endpoint().port_specifier_case() ==
+                     api::core::SocketAddress::PortSpecifierCase::kPortValue,
+                 "X-Ray daemon port must be specified as number. Not a named port.");
```

Comment:
These are actual config errors and shouldn't crash. Can you turn these into exceptions with appropriate tests?